src/CADL.Extension/Emitter.Csharp/src/lib/model.ts

```diff
@@ -250,8 +249,25 @@ export function getInputType(
             }
         }
 
-        if (intrinsicName && intrinsicName === "Map") {
-            return getInputTypeForMap(program, m);
+        // if (intrinsicName && intrinsicName === "Map") {
+        //     return getInputTypeForMap(program, m);
+        // }
+        if (m.indexer) {
+            if (isNeverType(m.indexer.key)) {
```

Comment:
Can we directly use the below directly?

if (!isNeverType(m.indexer.key)){
    do sth. 
}